lightning/src/util/test_utils.rs

```diff
@@ -194,10 +224,9 @@ impl<'a> chain::Watch<EnforcingSigner> for TestChainMonitor<'a> {
 }
 
 pub struct TestPersister {
-	pub update_ret: Mutex<chain::ChannelMonitorUpdateStatus>,
-	/// If this is set to Some(), after the next return, we'll always return this until update_ret
-	/// is changed:
-	pub next_update_ret: Mutex<Option<chain::ChannelMonitorUpdateStatus>>,
+	/// The queue of update statuses we'll return. If none are queued, ::Completed will always be
```

Comment:
>  Instead, we should consider tracking whether monitor update status results were set at all, and if they were, panic if we try to get a result and no results are available

To confirm, the goal with this would be to force tests to explicitly adapt to e.g. a new monitor update rather than having their behavior silently change?

Reply:
Hmm, so that causes 30 test failures. They don't look too difficult, but I hadn't realized how deep this refactor was and I think I can rewrite the tests to avoid needing it more easily than complete it. Thoughts on reverting this commit?

Reply:
Okay, I'll probably go through with the refactor in parallel to #1916 then. May not be able to follow up on this until sometime next week.